### PR TITLE
Add belongsToMany support for createAssociation

### LINKED ISSUE #20: Footprint createAssociation does not work correctly.
I have the following models:
User.js:
```
module.exports = class User extends Model {
  static config (app,Sequelize) {
    return {
       options: {
         hooks:{
          beforeCreate: (values,options) =>{
            return app.services.Utils.createBcryptHash(values.password).then(
              (hash)=>{
                values.password = hash
              },
              (err)=>{
                app.log.error(err)
              }
            )
          }
         },
         instanceMethods:{
          toJSON: function () {
            let $user = this.dataValues;
            delete $user.password;
            return $user;
          }
         },
         classMethods: {
           //If you need associations, put them here
           associate: (models) => {
             //More information about associations here : http://docs.sequelizejs.com/en/latest/docs/associations/
             models.User.belongsToMany(models.Role, {
               as: 'roles',
               onDelete: 'CASCADE',
               through:{
                model: models.UserRole,
                unique:false
               },
               foreignKey: "user_id"
             })
           }
         }
       }
     }
  }

  static schema (app,Sequelize) {
    return {
      name: {
        type: Sequelize.STRING,
        allowNull: true,
        unique : true
      },
      email:{
        type: Sequelize.STRING,
        allowNull: false,
        unique : true,
        validate:{
          isEmail : true
        }
      },
      external:{
        type: Sequelize.BOOLEAN
      },
      password: {
        type : Sequelize.STRING,
        allowNull: false
      },
      displayName: {
        type : Sequelize.STRING,
        allowNull: true
      }
    }
  }
}
```

Role.js

```
module.exports = class Role extends Model {

  static config () {
    return {
       options: {
         classMethods: {
           //If you need associations, put them here
           associate: (models) => {
             //More information about associations here : http://docs.sequelizejs.com/en/latest/docs/associations/
             models.Role.belongsToMany(models.User, {
               as: 'users',
               onDelete: 'CASCADE',
               through:{
                model: models.UserRole,
                unique:false
               },
               foreignKey: "role_id"
             })
           }
         }
       }
     }
  }

  static schema (app,Sequelize) {
    return {
      name: {
        type: Sequelize.STRING,
        allowNull: false,
        unique:true
      }
    }
  }
}
```

UserRole.js

```
module.exports = class UserRole extends Model {

  static config () {
  }

  static schema (app,Sequelize) {
    return {
      role_id: {
        type: Sequelize.INTEGER,
        references: {
          model: 'role',
          key: 'id'
        },
        allowNull: false
      },
      user_id:{
        type: Sequelize.INTEGER,
        references:{
          model: 'user',
          key: 'id'
        },
        allowNull: false
      }
    }
  }
}
```

when I access /POST /api/v1/user/1/roles,  body: {"name":"role"}

The application creates the role with name:role, but does not create the association (UserRole). This is the orm output:

```
Executing (b68a81ff-1f56-4c0e-bdb7-9c3be8215ec8): SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ;
Executing (b68a81ff-1f56-4c0e-bdb7-9c3be8215ec8): INSERT INTO "role" ("id","name","createdAt","updatedAt") VALUES (DEFAULT,'role','2016-06-14 19:32:25.273 +00:00','2016-06-14 19:32:25.273 +00:00') RETURNING *;
Executing (b68a81ff-1f56-4c0e-bdb7-9c3be8215ec8): UPDATE "user" SET "updatedAt"='2016-06-14 19:32:25.289 +00:00' WHERE "id" = '1'
Executing (b68a81ff-1f56-4c0e-bdb7-9c3be8215ec8): COMMIT;
```

### PR DESCRIPTION
**Pull request by PlasmaPower**
Resolves #20

### CODE REVIEW

File: api/services/FootprintService.js [hunk 1]
```diff
@@ -197,18 +198,28 @@ module.exports = class FootprintService extends Service {
...
         // Used for things like belongsTo
         if (!association.foreignKeyField) {
-          return this.update(parentModelName, parentId, {
-            [association.identifierField]: child.id
-          }, options).then(() => child)
-        }
-        else {
-          return child
+          promise = promise.then(this.update(parentModelName, parentId, {
+            [association.identifierField]: child[childModel.primaryKeyAttribute]
```
Comment (PlasmaPower) [inline api/services/FootprintService.js]: I've made a couple of small unrelated changes here, hopefully that's okay.

**Comment by jaumard** [comment]
@cesar2064 can you confirm this is working  ? Don't have time to test myself, the code look good to be :) so if it's working I merged it

**Comment by PlasmaPower** [comment]
@jaumard https://github.com/trailsjs/trailpack-sequelize/issues/20#issuecomment-226571804

> Thanks @PlasmaPower, your implementations work fine :)

I'm pretty sure they're tested. Also, you can look at the tests which pass.

**Comment by cesar2064** [comment]
Y es, I tested it and it worked fine.

**Comment by jaumard** [comment]
Good! thanks both of you ! :D
